resolver/dns_resolver.ml

```diff
@@ -78,19 +78,19 @@ let pick rng = function
   | [ x ] -> Some x
   | xs -> Some (List.nth xs (Randomconv.int ~bound:(List.length xs) rng))
 
-let build_query ?id ?(recursion_desired = false) t ts proto question retry zone edns ip =
+let build_query ?id ?(recursion_desired = false) ?(checking_disabled = false) ?(dnssec_ok = true) t ts proto question retry zone edns ip =
   let id = match id with Some id -> id | None -> Randomconv.int16 t.rng in
   let header =
-    (* TODO not clear about this.. *)
     let flags =
+      let flags = Packet.Flags.singleton `Checking_disabled in
```

Comment:
comes from an RFC, should refer to it -- otherwise it catches you by surprise -- maybe https://www.rfc-editor.org/rfc/rfc4035#section-4.6